Ginger/GingerTest/Variable Dependancies/VariableDependancyTest.cs

```diff
@@ -0,0 +1,224 @@
+            selectionList.OptionalValuesList.Add(new OptionalValue("b"));
+            selectionList.SelectedValue = selectionList.OptionalValuesList[0].Value;
+            selectionList.ItemName = "MyVar";
+            BF1.Variables.Add(selectionList);
+
+            //Added dependancies in activities
+            string[] variableValues = new string[] { "a", "b" };
+
+            VariableDependency actiVD0 = new VariableDependency(selectionList.Guid, selectionList.ItemName, variableValues);
+            activityList[0].VariablesDependencies.Add(actiVD0);
+
+            VariableDependency actiVD1 = new VariableDependency(selectionList.Guid, selectionList.ItemName, variableValues);
+            activityList[1].VariablesDependencies.Add(actiVD1);
+
+            VariableDependency actiVD2 = new VariableDependency(selectionList.Guid, selectionList.ItemName, new string[] { "b" });
+            activityList[2].VariablesDependencies.Add(actiVD2);
+            
+            //Act
+            mGR.RunBusinessFlow(BF1);
+
+            //Assert
+            Assert.AreEqual(BF1.Activities[0].Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+            Assert.AreEqual(BF1.Activities[1].Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+            Assert.AreEqual(BF1.Activities[2].Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Skipped);
+
+            //Changed the selected value of selection list 
+            ((VariableSelectionList)BF1.Variables[0]).SelectedValue = selectionList.OptionalValuesList[1].Value;
+
+            //Act
+            mGR.RunBusinessFlow(BF1);
+
+            //Assert
+            Assert.AreEqual(BF1.Activities[0].Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+            Assert.AreEqual(BF1.Activities[1].Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+            Assert.AreEqual(BF1.Activities[2].Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void ActionVariablesDependancyTest()
+        {
+            //Arrange
+            ObservableList<Activity> activityList = BF1.Activities;
+
+            Activity activity = activityList[0];
+            ObservableList<IAct> actionList = activity.Acts;
+
+            activity.EnableActionsVariablesDependenciesControl = true;
+
+            //Added selection list variable in activity
+            VariableSelectionList selectionList = new VariableSelectionList();
+            selectionList.OptionalValuesList.Add(new OptionalValue("abc"));
+            selectionList.OptionalValuesList.Add(new OptionalValue("xyz"));
+            selectionList.SelectedValue = selectionList.OptionalValuesList[1].Value;
+            selectionList.ItemName = "MyActVar";
+            activity.Variables.Add(selectionList);
+
+            //added action level dependancies
+            string[] variableValues = new string[] { "abc", "xyz" };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4817688)